conan/tools/files/files.py

```diff
@@ -520,3 +521,22 @@ def swap_child_folder(parent_folder, child_folder):
     child = os.path.join(parent_folder, child_folder)
     for f in os.listdir(child):
         shutil.move(os.path.join(child, f), os.path.join(parent_folder, f))
+
+
+def fix_apple_shared_install_name(conanfile):
+    def _osx_collect_dylibs(lib_folder):
+        return [f for f in os.listdir(lib_folder) if f.endswith(".dylib")
+                and not os.path.islink(os.path.join(lib_folder, f))]
+
+    def _fix_install_name(lib_name, lib_folder):
+        command = "install_name_tool -id @rpath/{} {}".format(lib_name, os.path.join(lib_folder,
+                                                                                     lib_name))
+        conanfile.run(command)
+
+    if conanfile.settings.get_safe("os") == "Macos" and conanfile.options.get_safe("shared", False):
```

Comment:
Shouldn't it be for any Apple OS?

```suggestion
    if is_apple_os(conanfile.settings.get_safe("os")) and conanfile.options.get_safe("shared", False):
```